crates/executor/src/executor.rs

```diff
@@ -195,6 +195,10 @@ where
                 new_account.account_state = if account.storage_cleared {
                     new_account.storage.clear();
                     AccountState::StorageCleared
+                } else if matches!(new_account.account_state, AccountState::StorageCleared) {
```

Comment:
it is a bit confusing that `account.storage_cleared` and `new_account.account_state == AccountState::StorageCleared` are different states